amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/KinesisClientLibConfigurator.java

```diff
@@ -69,8 +71,19 @@ public MultiLangDaemonConfiguration getConfiguration(Properties properties) {
         });
 
         Validate.notBlank(configuration.getApplicationName(), "Application name is required");
-        Validate.notBlank(configuration.getStreamName(), "Stream name is required");
+
+        if (configuration.getStreamArn() != null && !configuration.getStreamArn().trim().isEmpty()) {
+            final Arn streamArnObj = Arn.fromString(configuration.getStreamArn());
+            StreamIdentifier.validateArn(streamArnObj);
+            //Parse out the stream Name from the Arn (and/or override existing value for Stream Name)
+            final String streamNameFromArn = streamArnObj.resource().resource();
+            configuration.setStreamName(streamNameFromArn);
+
+        } else {
+            Validate.notBlank(configuration.getStreamName(), "Stream name or Stream Arn is required. Stream Arn takes precedence if both are passed in.");
```

Comment:
Looks like we can eliminate this `else` branch. Rather than assume an ARN will provide a non-blank stream name, we could check `Validate.notBlank(...)` regardless.

Reply:
Thats fair, i figured if it was being set in line 80 that this was redundant. But we can simplify it.